akka-http2-support/src/main/scala/akka/http/impl/engine/http2/rendering/HttpResponseHeaderHpackCompression.scala

```diff
@@ -85,29 +68,18 @@ final class HttpResponseHeaderHpackCompression extends GraphStage[FlowShape[Http
           encoder.encodeHeader(os, nameBytes, valueBytes, false)
         }
 
-      // copy buffer to ByteString
-      mkByteString(buf)
+      val res = ByteString(os.toByteArray)
+      os.reset()
```

Comment:
yeah, let's just use the `os` here, thanks for noticing; we'll see if it matters perf wise thtough